drivers/adc/adc_ite_it8xxx2.c

```diff
@@ -112,10 +109,14 @@ static int adc_it8xxx2_channel_setup(const struct device *dev,
 		return -EINVAL;
 	}
 
-	/* The channel is set to ADC alternate function */
-	pinmux_pin_set(config[channel_id].pinctrls,
-		       config[channel_id].pin,
-		       config[channel_id].alt_fun);
+	/* Set the pin to ADC alternate function. */
+	status = pinctrl_configure_pins(&config->pcfg->states->pins[channel_id],
+					1, 0);
```

Comment:
Use PINCTRL_REG_NONE instead of 0

Reply:
pinctrl states are opaque, so drivers can’t make use of encoded information within states. Please find an alternative.

Reply:
Use pinctrl_apply_state function instead of pinctrl_configure_pins. And if the pins need to set the adc alternate function, pinctrl-0 is determined by the board dts.